openedx/core/djangoapps/notifications/tasks.py

```diff
@@ -115,32 +121,26 @@ def send_notifications(user_ids, course_key: str, app_name, notification_type, c
         if not preferences:
             continue
 
+        notifications = []
         for preference in preferences:
-            preference = update_user_preference(preference, preference.user_id, course_key)
-            if not (
+            user_id = preference.user_id
+            preference = update_user_preference(preference, user_id, course_key)
```

Comment:
[question] - why do we need this updation here?

Reply:
It updates user preference to latest preference version.